test/integration_test/migration_features_test.go

```diff
@@ -330,9 +349,328 @@ func testMigrationStashStrategyWithStartApplication(t *testing.T) {
+	mig1, err := asyncdr.CreateMigration(firstMigrationName, appData.Ns, clusterPairName, appData.Ns, &includeVolumesFlag, &includeResourcesFlag, &startApplications)
+	err = asyncdr.WaitForMigration([]*storkapi.Migration{mig1})
+	require.NoError(t, err, "Error waiting for migration")
+	logrus.Infof("Migration %s/%s completed successfully ", appData.Ns, mig1.Name)
+
+	// get the resource UID to verify if it is getting overwritten
+	initialCMUIDs, err := getStashedCMUIDs(appData.Ns)
+	require.NoError(t, err, "Error getting stashed configmap UIDs")
+
+	// Do the migration again and verify the resource ID
+	secondMigrationName := fmt.Sprintf("%s%s-%d", migNamePref, appName, 2)
+	logrus.Infof("Starting migration %s/%s with startApplication false, iteration number: 2", appData.Ns, secondMigrationName)
+	mig2, err := asyncdr.CreateMigration(secondMigrationName, appData.Ns, clusterPairName, appData.Ns, &includeVolumesFlag, &includeResourcesFlag, &startApplications)
+	err = asyncdr.WaitForMigration([]*storkapi.Migration{mig2})
+	require.NoError(t, err, "Error waiting for migration")
+	logrus.Infof("Migration %s/%s completed successfully ", appData.Ns, mig2.Name)
+
+	// get the resource UID to verify if it is getting overwritten
+	finalCMUIDs, err := getStashedCMUIDs(appData.Ns)
+	require.NoError(t, err, "Error getting stashed configmap UIDs")
+
+	require.Equal(t, initialCMUIDs, finalCMUIDs, "stashed configmap UIDs are not matching")
+
+	err = setDestinationKubeConfig()
+	require.NoError(t, err, "Error setting dest kubeconfig")
+
+	// validate CRD on destination
+	err = asyncdr.ValidateCRD(appData.ExpectedCrdList, destClusterConfigPath)
+	require.NoError(t, err, "Error validating destination crds")
+
+	// validate that CR is not present as it is stashed
+	validated, err := validateCR(appName, appData.Ns, destClusterConfigPath)
+	require.NoError(t, err, fmt.Sprintf("Error validating CR for app %s in namespace %s", appName, appData.Ns))
+	require.False(t, validated, fmt.Sprintf("CR for app %s in namespace %s should not have been present", appName, appData.Ns))
+
+	// validate stashed cm in destination
+	stashedCMCount := getStashedConfigMapCount(t, appData.Ns)
+	operatorCRMap := getSupportedOperatorCRMapping()
+	if _, ok := operatorCRMap[appName]; !ok {
+		require.True(t, ok, fmt.Sprintf("app %s is not currently supported in test framework", appName))
+	}
+	expectedStashedCMCount := len(operatorCRMap[appName])
+	require.Equal(t, expectedStashedCMCount, stashedCMCount, fmt.Sprintf("expected stashed configmap count %d got %d", expectedStashedCMCount, stashedCMCount))
+
+	// activate app
+	err = activateAppUsingStorkctl(appData.Ns, false)
+	require.NoError(t, err, fmt.Sprintf("Error activating app in namespace %s", appData.Ns))
+
+	err = setDestinationKubeConfig()
+	require.NoError(t, err, "Error setting dest kubeconfig")
+
+	logrus.Infof("Waiting for application pods to come up")
+	time.Sleep(5 * time.Minute)
```

Comment:
Should we poll for app status and move forward if app is up instead of waiting for 5 mins?

Reply:
yes, will add that in another PR.